src/components/application_manager/src/application_impl.cc

```diff
@@ -526,6 +537,10 @@ void ApplicationImpl::WakeUpStreaming(
   LOG4CXX_AUTO_TRACE(logger_);
 
   if (ServiceType::kMobileNav == service_type) {
+    if (!video_streaming_approved()) {
```

Comment:
One potential solution, add an if statement which shuts down an existing stream if it is present when this method is called:

streamer_adapter.cc
```cpp
int32_t StreamerAdapter::get_current_app_id() {
  return current_application_;
}
```

media_manager_impl.cc
```cpp
int32_t MediaManagerImpl::GetCurrentStreamingApp(protocol_handler::ServiceType service_type) {
  return streamer_[service_type]->get_current_app_id();
}
```

application_manager_impl.cc
```cpp
ApplicationSharedPtr ApplicationManagerImpl::GetCurrentStreamingApp(protocol_handler::ServiceType service_type) {
  return application(media_manager_->GetCurrentStreamingApp(service_type));
}
```

ApplicationImpl::WakeUpStreaming
```cpp
ApplicationSharedPtr current_streaming_app = application_manager_.GetCurrentStreamingApp(service_type);
if (current_streaming_app->app_id() != 0 && current_streaming_app->app_id() != app_id()) {
  current_streaming_app->StopStreamingForce(service_type);
}
```